tests/unit/model-test.js

```diff
@@ -164,6 +164,82 @@ test("isDirty on the individual property is reset after save", function(assert){
     assert.equal(undefined, brandon.get("firstNameIsDirty"));
 });
 
+test("isDirty on the model is reset only after all values set back to original values", function(assert){
+    brandon = Person.create(data);
+    assert.equal("Brandon", brandon.get("firstName"));
+    assert.equal("Williams", brandon.get("lastName"));
+    assert.equal(false, brandon.get("isDirty"));
+    brandon.set("firstName", "foo");
+    brandon.set("lastName", "bar");
+    assert.equal(true, brandon.get("isDirty"));
+    brandon.set("firstName", "Brandon");
+    assert.equal(true, brandon.get("isDirty"));
+    brandon.set("lastName", "Williams");
+    assert.equal(false, brandon.get("isDirty"));
+});
+
+test("isDirty on the model is reset after value set back to original value", function(assert){
+    brandon = Person.create(data);
+    assert.equal("Brandon", brandon.get("firstName"));
+    assert.equal(false, brandon.get("isDirty"));
+    brandon.set("firstName", "baz");
+    assert.equal(true, brandon.get("isDirty"));
+    brandon.set("firstName", "Brandon");
+    assert.equal(false, brandon.get("isDirty"));
+});
+
+test("isDirty on the model is reset when original value is empty string and set back to empty string", function(assert){
+    brandon = Person.create({id: 1, firstName: "", lastName: "Williams"});
+    assert.equal(undefined, brandon.get("firstName"));
+    assert.equal(false, brandon.get("isDirty"));
```

Comment:
After some playing around with the validation jsbin I'm starting to regret what I said above. Originally I thought we'd simply update the validation library to be "smarter" as this change seemed correct (in the identity map itself). But two negative things happened as I started to go down that road

First up -from a real app (end user) point of view. When I open a form and simply "tab into" or "click" an input the model is suddenly "dirty" or "changed" .... yet the user hasn't done anything at all. In fact (from a UX perspective) it's annoying to tab between fields and have every single validation suddenly fire on you without typing anything (in my opinion).

Second - the validation of the component got complex quickly because in addition to checking to see if isDirty was true I need to see if the models value (for that given field) was a typeof string -and if yes if the value was "" (to determine if I should hide/not hide the errors)

I'm thinking we should leave the behavior as is because the tradeoffs on the other side are worse than this seemingly bad/incorrect/broken inconsistency. This feels like the first "wart" in the project so it kills me to even suggest we leave it as is but without changing how a setter is called on a model bound input I don't see any other way around this currently.

Open for ideas/suggestions before we pull the trigger